presto-main/src/main/java/com/facebook/presto/sql/relational/RowExpressionMatcher.java

```diff
@@ -0,0 +1,294 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.relational;
+
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.RowBlock;
+import com.facebook.presto.spi.relation.CallExpression;
+import com.facebook.presto.spi.relation.ConstantExpression;
+import com.facebook.presto.spi.relation.InputReferenceExpression;
+import com.facebook.presto.spi.relation.LambdaDefinitionExpression;
+import com.facebook.presto.spi.relation.RowExpression;
+import com.facebook.presto.spi.relation.RowExpressionVisitor;
+import com.facebook.presto.spi.relation.SpecialFormExpression;
+import com.facebook.presto.spi.relation.VariableReferenceExpression;
+import com.facebook.presto.spi.type.ArrayType;
+import com.facebook.presto.spi.type.RowType;
+import com.facebook.presto.spi.type.Type;
+import com.google.common.collect.ImmutableMap;
+import io.airlift.slice.DynamicSliceOutput;
+import io.airlift.slice.Slice;
+import io.airlift.slice.SliceOutput;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import static java.lang.Math.toIntExact;
+
+public class RowExpressionMatcher
+{
+
+    public boolean match(RowExpression actual, RowExpression expected)
+    {
+        return actual.accept(new RowExpressionMatchVisitor(), expected);
+    }
+
+    public boolean match(RowExpression actual, RowExpression expected, Map<String, String> aliases)
+    {
+        return actual.accept(new RowExpressionMatchVisitor(aliases), expected);
+    }
+
+    public class RowExpressionMatchVisitor
+            implements RowExpressionVisitor<Boolean, RowExpression>
+    {
+        private Map<String, String> aliases;
+
+        public RowExpressionMatchVisitor()
+        {
+            this.aliases = new HashMap<>();
+        }
+
+        public RowExpressionMatchVisitor(Map<String, String> aliases)
+        {
+            this.aliases = aliases;
```

Comment:
Make a copy instead, so that we will not accidentally change the passed in map. Also add `requireNonNull` to  make sure aliases is not null.